.github/workflows/ci.yml

```diff
@@ -252,13 +267,77 @@ jobs:
           name: nuget-packages
           path: ./Bin/Release/ILGPU*.${{ steps.package.outputs.version }}.*nupkg
 
-    outputs:
-      version: ${{ steps.version.outputs.version }}
+  # Create the Samples zip file, replacing ILGPU project references with equivalent NuGet package references.
+  package-samples:
+    needs: check-version
+    runs-on: windows-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup the latest .NET 5 SDK
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 5.0.x
+
+      # Change the ILGPU project references to NuGet package references
+      - name: Update Sample References
+        id: package
+        run: |
+          $version = "${{ needs.check-version.outputs.version }}"
+
+          if (-not ("${{ github.ref }}" -like "refs/tags/v*")) {
+            $suffix = "preview-$(Get-Date -Format yyyyMMddHHmmss -AsUTC)-$(git rev-parse --short HEAD)"
+            $params = "--version-suffix", $suffix
+            $version = "$version-$suffix"
+          }
+
+          # NB: Perform operations in the Samples folder, so that the relative paths returned by
+          # the dotnet CLI are correct.
+          pushd Samples
+
+          # Remove ILGPU libraries from Samples solution.
+          $projectPaths = dotnet sln  list | Select-String -Pattern "\.\.\\Src\\"
+          dotnet sln remove $projectPaths
+
+          # Update each sample project individually.
+          $projectPaths = dotnet sln list | Select-String -Pattern "\.csproj$"
+
+          ForEach ($projectPath in $projectPaths) {
+            # Replace existing ILGPU libaries with NuGet package reference.
+            $referencePaths = dotnet list $projectPath reference | Select-String -Pattern "\.\.\\Src\\"
+            ForEach ($referencePath in $referencePaths) {
+              dotnet remove $projectPath reference $referencePath
+
+              $found = $referencePath -match '\\([^\\]*).csproj$'
+              if ($found) {
+                $libraryName = $matches[1]
+                dotnet add $projectPath package $libraryName -v $version --no-restore
+              }
+            }
+          }
```

Comment:
possible improvement: I think it would be useful to validate that the samples projects build cleanly after this change.
We could do this by:
* depending on `package-library`
* downloading the NuGet assets
* adding them to a local repo
* adding the local repo to a local `nuget.config`
* building the solution (after having replaced all the project references)

Reply:
I have added a new `verify-samples` job that downloads both artifacts, to ensure that ILGPU source code cannot be found. This is necessary because I found a bug that the F# sample project did not have it's project references changed, but everything was still compiling because the ILGPU source code could still be found.

Reply:
OK, I changed the `samples` artifact to include a local `nuget.config`, and the ILGPU nuget packages. This will make the `samples` artifact a standalone zip, without having to rely on the user manually added FeedzIO to their nuget package configuration.

Reply:
Once we solve issue #589, I feel that we should _not_ include the ILGPU nuget packages in the `samples` artifact.
Instead I would suggest we add a `nuget.config` pointing to FeedzIO when building a preview, and no config at all when building a release. As the packages would not have yet been published, we would download them in `verify-samples` and add them to a local source before building. When a user tries to use the `samples` artifact, it would just work for them as the packages will be found on NuGet/FeedzIO.

Reply:
The one issue I have with creating a `nuget.config` that points to FeedzIO is that PR builds do not upload their nuget packages - this would prevent the `samples` artifact on these PR builds from compiling.

Perhaps we should disable creating the `samples` artifact on PR builds?

Reply:
Agreed. I don't think that it makes sense or would be useful to anyone to download the `samples` artifact for a PR build.

Reply:
OK, updated the CI pipeline - `verify-samples` has been merged into `package-samples` so that we can selectively upload the sample artifacts. On preview releases, we will create `nuget.config` with a FeedzIO source.